input/otlp/metrics.go

```diff
@@ -125,23 +125,42 @@ func (c *Consumer) convertResourceMetrics(
 	}
 	scopeMetrics := resourceMetrics.ScopeMetrics()
 	for i := 0; i < scopeMetrics.Len(); i++ {
-		c.convertScopeMetrics(scopeMetrics.At(i), baseEvent, timeDelta, out, remainingDataPoints, remainingMetrics)
+		c.handleScopeMetrics(scopeMetrics.At(i), resource, baseEvent, timeDelta, out, remainingDataPoints, remainingMetrics)
 	}
 	return
 }
 
-func (c *Consumer) convertScopeMetrics(
+func (c *Consumer) handleScopeMetrics(
 	in pmetric.ScopeMetrics,
+	resource pcommon.Resource,
 	baseEvent *modelpb.APMEvent,
 	timeDelta time.Duration,
 	out *modelpb.Batch,
 	remainingDataPoints, remainingMetrics *int64,
 ) {
 	ms := make(metricsets)
+	// Add the original otel metrics to the metricset.
 	otelMetrics := in.Metrics()
 	for i := 0; i < otelMetrics.Len(); i++ {
 		c.addMetric(otelMetrics.At(i), ms, remainingDataPoints, remainingMetrics)
 	}
+	// Handle remapping if any. Remapped metrics will be added to a new
+	// metric slice and then processed as any other metric in the scope.
+	// TODO (lahsivjar): Possible to approximate capacity of the slice?
```

Comment:
Is it possible to count the metrics in `in`? If yes would multiply them by `c.remappers` produce a decently accurate figure?

Reply:
> Is it possible to count the metrics in in?

Unfortunately no, the remapping library has full control over which metrics will be remapped and thus the count of the remapped slice. We can get an upper bound but in most cases, it would be too big. I was thinking of adding a method in the remappers to get an approximate size (which each remapper can give) but maintaining such code would be a PITA as adding any new metric would require to update the count. So, TL;DR I don't have a good approach yet.

Reply:
Do we expect a high volume of hits? I wouldn't think so - so it probably doesn't matter much.

Reply:
> Do we expect a high volume of hits? I wouldn't think so - so it probably doesn't matter much.

Agreed.